### PR TITLE
Validations for employee/branch models have been added

### LINKED ISSUE #24: Data validation in models (Employee and Branches)
# REQUIREMENTS
 Employee model:
  - Not null values
  -  Unique private code and email

# DESIGN
Not necessary

### PR DESCRIPTION
**Pull request by brocha05**
# Description
What did you implemented/Why did you implemented this:
 - I added validations for employees/branch models

 - Why?
 - We dont want employees or branches with empty fields, also, some data should not be repeated.

 # Testing
Just try to add a employee with a empty field or with an email that is been using in another employee.

### CODE REVIEW

**Review by dannharo — COMMENTED** [review]
add tests please

**Review by YilKanda — CHANGES_REQUESTED** [review]
Resolve conflicts in your branch.

**Review by vyk2rr — DISMISSED** [review]
Add unit test to Models branch and Employee

**Comment by brocha05** [comment]
@vyk2rr @YilKanda Could you check again this PR? Thank you

File: spec/models/branch_spec.rb [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+
+  it 'Branch name must be unique' do
+    @branch.save!
+    branch2 = Branch.new(name: 'Recursos humanos', address: 'Manzanillo, Colima')
+    expect(branch2).not_to be_valid
+  end
+
+
+
+
```
Comment (AlejandroCuriel) [inline spec/models/branch_spec.rb]: you don't need to have so many line breaks

**Review by juperameza — DISMISSED** [review]
LGTM! :) <3

**Review by AlejandroCuriel — DISMISSED** [review]
LGTM :rocket:  :racehorse:

**Review by juperameza — APPROVED** [review]
LGTM x2

**Review by AlejandroCuriel — APPROVED** [review]
LGTM :rocket: :racehorse:

**Review by YilKanda — APPROVED** [review]
LGTM!

**Comment by vyk2rr** [comment]
Great! you added the Rspec setup and tests, I have a little feedback for you:

based on this reading: https://www.betterspecs.org

------
This is the test you added: 

```ruby
escribe Employee do
  before(:each) do
    @employee = Employee.new(...
  end

  it 'is valid create with all fields values' 
  it 'Name field should have presence' 
  it 'Email field should have presence'
  it 'Position field should have presence' 
  it 'private_code field should have presence' 
  it 'Active field should have presence' 
  it 'Branch_id field should have presence' 
  it 'Email must be unique' 
  it 'Private_code must be unique' 
  it 'Private_code should have 6 characters' 
end
```

I took that and re-created that in a way that it is easier for you to read and understand:

```ruby
escribe Employee do
  before(:each) do
    @employee = Employee.new(...
  end

  it 'creates employees' 
  it "validates Employee's Name presence" 
  it "validates Employee's Email presence" 
  it "validates Employee's Position presence" 
  it "validates Employee's Private Code presence" 
  it "validates Employee's Active field presence" 
  it "validates Employee's branch_id field presence" 
  it "validates Employee's email presence" 
  it "validates Employee's email uniqueness" 
  it "validates Employee's private code uniqueness" 
  it "validates Employee's private_code with the right structure"
end
```

But even better you can take advantage of 'context' and we group tests, something more like this:

```ruby
escribe Employee do
  before(:each) do
    @employee = Employee.new(...
  end

  it 'creates employees with valid data' 
  it 'creates employees with missing data' do 
    # validates presence of: Name, Email, Position, Private Code, Active field, branch_id field

    # ^ in here you try to create an Employee with empty values
    # ^ then you look at the errors of active record:
    #     - all errors should be there for all the fields you are missing
    #     (just make sure they are there, and that's it)
  end

  it "validates Employee's email uniqueness"
  it "validates Employee's private code uniqueness"

  context 'having wrong values' do
    it "validates Employee's private_code with the right structure"
  end
end
```

I hope that it helps! :) I know this is already approved, if you read me let me know What do you think?

**Pull request merged**